lib/honeydew/node_monitor_supervisor.ex

```diff
@@ -1,20 +1,23 @@
 defmodule Honeydew.NodeMonitorSupervisor do
   alias Honeydew.NodeMonitor
 
-  def start_link(queue, nodes) do
-    import Supervisor.Spec
-
-    children = [worker(NodeMonitor, [], restart: :transient)]
+  def child_spec([queue, _nodes] = opts) do
+    %{
+      id: Honeydew.supervisor(queue, :node_monitor),
+      start: {__MODULE__, :start_link, opts}
+    }
+  end
 
+  def start_link(queue, nodes) do
     opts = [
       name: Honeydew.supervisor(queue, :node_monitor),
-      strategy: :simple_one_for_one
+      strategy: :one_for_one
```

Comment:
i'm pretty sure `:one_for_one` is the default for `DynamicSupervisor`s, don't think we need to be explicit.

Reply:
(also in various other places in the PR)

Reply:
It appears to be required: https://github.com/elixir-lang/elixir/blob/v1.6.6/lib/elixir/lib/dynamic_supervisor.ex#L477